app/controllers/api/v1_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Api::V1Controller < ApiController
+  before_action :authenticate
+
+  protected
+
+  def authenticate
+    @current_user = token_auth if params[:auth_token]
+    @current_user ||= basic_auth
+
+    render_unauthorized "Invalid API Credentials" unless @current_user
+  end
+
+  def render_unauthorized message
+    payload = { errors: [ { detail: message } ] }
+    render json: payload, status: :unauthorized
```

Comment:
it's a pity Rails doesn't `throw` & `catch` here, isn't it :\

Reply:
yeah :/